app/controllers/financialStatement/ChargeDetailsController.scala

```diff
@@ -33,45 +34,56 @@ import utils.DateHelper.{dateFormatterDMY, dateFormatterStartDate}
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class ChargeDetailsController @Inject()(
-                                         identify: IdentifierAction,
-                                         override val messagesApi: MessagesApi,
-                                         val controllerComponents: MessagesControllerComponents,
-                                         fsConnector: FinancialStatementConnector,
-                                         penaltiesService: PenaltiesService,
-                                         schemeService: SchemeService,
-                                         renderer: Renderer
+class ChargeDetailsController @Inject()(appConfig: FrontendAppConfig,
+                                        identify: IdentifierAction,
+                                        override val messagesApi: MessagesApi,
+                                        val controllerComponents: MessagesControllerComponents,
+                                        fsConnector: FinancialStatementConnector,
+                                        penaltiesService: PenaltiesService,
+                                        schemeService: SchemeService,
+                                        renderer: Renderer
                                        )(implicit ec: ExecutionContext)
   extends FrontendBaseController
     with I18nSupport
     with NunjucksSupport {
 
-  def onPageLoad(srn: String, startDate: LocalDate, chargeReference: String): Action[AnyContent] = identify.async {
+  def onPageLoad(identifier: String, startDate: LocalDate, chargeReference: String): Action[AnyContent] = identify.async {
     implicit request =>
-      schemeService.retrieveSchemeDetails(request.psaId.id, srn).flatMap { schemeDetails =>
-        fsConnector.getPsaFS(request.psaId.id).flatMap { psaFS =>
-          val filteredPsaFS = psaFS.filter(_.chargeReference == chargeReference)
+      fsConnector.getPsaFS(request.psaId.id).flatMap {
+        psaFS =>
+          val commonJson = Json.obj(
+            "heading" -> heading(psaFS.filter(_.chargeReference == chargeReference).head.chargeType.toString),
+            "isOverdue" -> penaltiesService.isPaymentOverdue(psaFS.filter(_.chargeReference == chargeReference).head),
+            "period" -> msg"penalties.period".withArgs(startDate.format(dateFormatterStartDate),
+              getQuarter(startDate).endDate.format(dateFormatterDMY)),
+            "chargeReference" -> psaFS.filter(_.chargeReference == chargeReference).head.chargeReference,
+            "list" -> penaltiesService.chargeDetailsRows(psaFS.filter(_.chargeReference == chargeReference).head)
+          )
 
-          if (filteredPsaFS.nonEmpty) {
+          if (psaFS.exists(_.chargeReference == chargeReference)) {
```

Comment:
This equality check is in the code twice. Could we not stick with original filter and check for empty instead? That feels a little less cluttered to me.

Reply:
Done